protostar/starknet_gateway/account_manager.py

```diff
@@ -0,0 +1,62 @@
+from starknet_py.net.signer import BaseSigner
+from starknet_py.net import AccountClient, KeyPair
+from starknet_py.net.gateway_client import GatewayClient
+from starknet_py.net.client_models import Call as SNCall
+
+from protostar.starknet import Address
+from protostar.starknet_gateway.gateway_facade import Fee
+from protostar.starknet_gateway.multicall.multicall_protocols import (
+    UnsignedMulticallTransaction,
+)
+
+from .multicall import SignedMulticallTransaction, MulticallAccountManagerProtocol
+
+
+class AccountManager(MulticallAccountManagerProtocol):
+    def __init__(
+        self,
+        private_key: int,
+        address: Address,
+        signer: BaseSigner,
+        gateway_url: str,
+        max_fee: Fee,
+    ):
+        self._private_key = private_key
+        self._signer = signer
+        self._max_fee = max_fee
+        gateway_client = GatewayClient(gateway_url)
+        self._account_client = AccountClient(
+            address=int(address),
+            client=gateway_client,
+            signer=signer,
+            key_pair=KeyPair.from_private_key(private_key),
+            supported_tx_version=1,
+        )
+
+    def get_account_address(self):
```

Comment:
we could have a public property named `account_address`, set in the constructor instead.

Reply:
1) I'd have to define an abstract property in the `MulticallAccountManagerProtocol` what creates awkward code. The use case class uses `AccountManager` indirectly through an interface to break a cyclic dependency.
2) I try to avoid creating hybrid objects. 

> Hybrids are classes that acts both as Data Structures (having public accessors and mutators)  and Objects (functions that do significant things)